### DISCUSSION TITLE
3.3.0-rc1

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#6086] [Design & tech]
## What's Changed
* Refactor Workflow Runtimes by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5444
* Multitenancy by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5159
* Flowchart variables are not serialized by FlowchartJsonConverter by @bobhauser in https://github.com/elsa-workflows/elsa-core/pull/5533
* Allow user to use DbContext pooling with Quartz by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/5625
* Register scheduler as singleton to avoid multiple starts of same task by @4lexKislitsyn in https://github.com/elsa-workflows/elsa-core/pull/5687
* fix(multitenancy): add missing TenantAgnostic flag to trigger store by @jeanbaptistedalle in https://github.com/elsa-workflows/elsa-core/pull/5689
* Respect Timeout and FailedToConnect in SendHttpRequestActivityResolver by @cschulznethaus in https://github.com/elsa-workflows/elsa-core/pull/5632
* Fixing some analyzer warnings by @lahma in https://github.com/elsa-workflows/elsa-core/pull/5617
* EF Core Oracle provider by @benjamin-hui in https://github.com/elsa-workflows/elsa-core/pull/5637
* Use database name from options when specified by @4lexKislitsyn in https://github.com/elsa-workflows/elsa-core/pull/5634
* feat: Modify the access level of FlowScope type and add Remove to specify the activity method by @Nokecy in https://github.com/elsa-workflows/elsa-core/pull/5752
* Add Bookmark Queue and Restore Background Activity Execution by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5758
* Enable Proto Actor Tracing for TraceLens by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5800
* Replace custom webhooks implementation with WebhooksCore by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5811
* Add OpenTelemetry module by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5810
* Drop NET 7 support and cleanup NuGet package references by @lahma in https://github.com/elsa-workflows/elsa-core/pull/5828
* Proto.Actor implementation for ChangeTokenSignalPublisher by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5817
* Upgrade to Jint 4 by @lahma in https://github.com/elsa-workflows/elsa-core/pull/5834
* Add Orchard Core Integration + Agents Module by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5871
* Fix background activity completion by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5882
* [Fix] Replace "bundles" with "apps" in the content of README.md to al… by @zergmk2 in https://github.com/elsa-workflows/elsa-core/pull/5885
* [Fix] DropIns Hosted Service File System Watcher by @RenatoCapelo in https://github.com/elsa-workflows/elsa-core/pull/5892
* Modified POST Execute API to Display Body in Swagger and Fixed Type-Casting Error in GET Execute API by @cvijayak in https://github.com/elsa-workflows/elsa-core/pull/5858
* Added support to retrieve the workflow details through Liquid script by @cvijayak in https://github.com/elsa-workflows/elsa-core/pull/5909
* Fix NullReferenceException when parsing the HttpResponse in the SendHttpRequestbase by @gamaSantos in https://github.com/elsa-workflows/elsa-core/pull/5930
* Set event payload when event bookmark resumes by @JaFonz in https://github.com/elsa-workflows/elsa-core/pull/5934
* Add Variable available in dropdown when configuring a activity input by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5918
* Add Agents Module with Semantic Kernel Integration by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5937
* Bump actions/download-artifact from 3 to 4.1.7 in /.github/workflows by @dependabot in https://github.com/elsa-workflows/elsa-core/pull/5941
* Fix : Change to parameter after update to Elsa.Studio packages to version 3.3.0-preview.524 by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5945
* Enh/5943 DropIns - Prevention of File Locks and Unloading by @RenatoCapelo in https://github.com/elsa-workflows/elsa-core/pull/5944
* [chore] Update upload-artifact major version for CI  by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5947
* Simplify Workflow Variables with JS by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5946
* Remove obsolete package MassTransit.Extensions.DependencyInjection by @DnikitaA in https://github.com/elsa-workflows/elsa-core/pull/5955
* Upgrade to NUKE 8.1 by @lahma in https://github.com/elsa-workflows/elsa-core/pull/5952
* Secrets API by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5967
* Upgrade Quartz and some warning fixes by @lahma in https://github.com/elsa-workflows/elsa-core/pull/5949
* Implement Activity State Filtering and JavaScript Integration by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5993
* fix: issue fixed for triggering correct workflow definition versionId by @sagargolu in https://github.com/elsa-workflows/elsa-core/pull/5979
* feat: add ITasksApi to Elsa.Api.Client by @cfgiugliano in https://github.com/elsa-workflows/elsa-core/pull/5995
* feat: register ITasksApi in AddDefaultApiClients by @cfgiugliano in https://github.com/elsa-workflows/elsa-core/pull/5999
* Fix: missing tenantAgnostic parameter passing by @yusianglin11010 in https://github.com/elsa-workflows/elsa-core/pull/6002
* Enhance logging by adding logger state to activity and workflow pipelines by @bobhauser in https://github.com/elsa-workflows/elsa-core/pull/5985
* Update System.Text.Json version reference by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6019
* Refactor Tenant Resolution to Use Async Local Storage for Operation-wide Access by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6022
* Bump System.Text.Json from 7.0.4 to 8.0.5 in /build by @dependabot in https://github.com/elsa-workflows/elsa-core/pull/6021
* Add db providers for secrets features by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/6020
* Implement multitenant HTTP routing by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6031
* Improved multitenancy support for HTTP workflows with per-tenant DbContext by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6032
* Manually register SerializedKeyValuePair.Key to BsonClassMap to prevent index creation failure by @yusianglin11010 in https://github.com/elsa-workflows/elsa-core/pull/6036
* Add Functionality to Copy Entire Path from HTTP Endpoint Activity #133 by @abhishekp21 in https://github.com/elsa-workflows/elsa-core/pull/6037
* Add missing DefaultHttpEndpointBasePathProvider registeration by @yusianglin11010 in https://github.com/elsa-workflows/elsa-core/pull/6040
* Add JsonNode BSON serializer by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6043
* Enhance error handling in `PolymorphicObjectConverter` by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6051
* Fix Bulk Dispatch Missing Input by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6052
* Include $schema when exporting workflow definition by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6056
* Introduce Log Persistence Strategy by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6057
* Update packages by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6058
* Add multitenancy support for background tasks by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6059
* Retention Module by @Sverre-W in https://github.com/elsa-workflows/elsa-core/pull/5344
* Refactor JS expressions for accessing secrets by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6070
* Refactor FormUrlEncodedHttpContentFactory for better flexibility by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6074
* Add workflow cancellation notifications by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6075
* Add workflow activation and startup services. by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6076
* Add bookmark queue purger functionality by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6080
* Enable CPM transitive pinning and resolve version conflicts by @lahma in https://github.com/elsa-workflows/elsa-core/pull/6069

## New Contributors
* @bobhauser made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5533
* @cschulznethaus made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5632
* @benjamin-hui made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5637
* @RenatoCapelo made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5892
* @gamaSantos made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5930
* @JaFonz made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5934
* @DnikitaA made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5955
* @sagargolu made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5979
* @cfgiugliano made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5995
* @yusianglin11010 made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/6002
* @abhishekp21 made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/6037

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/3.2.1...3.3.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/3.3.0-rc1'>3.3.0-rc1</a>.</em>

**Comment by yusianglin11010** [comment]
I'm really excited about the upcoming multitenancy feature. I was wondering if it would be possible to access a pre-release version of the package through NuGet or Feedz?

**Reply by sfmskywalker** [reply]
Yes, as soon as I ave repaired the broken build, version 3.3.0-rc1 will be available from NuGet and Feedz. Probably later today.

**Reply by sfmskywalker** [reply]
Here you go: https://www.nuget.org/packages/Elsa/3.3.0-rc1
